### PR TITLE
Adds the ability for multiple events with promises to be tracked and …

### PR DESCRIPTION
**Pull request by johnspackman**
…awaited on

### CODE REVIEW

**Review by derrell — CHANGES_REQUESTED** [review]
Can you explain a bit of what this is doing? It's in a mixin that will likely cause it to have wide-ranging consequences so I'd like to fully understand it.

**Comment by johnspackman** [comment]
events can be asynchronous, and can themselves trigger actions which are asynchronous - when they do so, they return a Promise.  If you're just firing one event (and there are no side effects) then it is easy to await firing, but lots of code does not do that.  This fix tracks any promises returned from an event being fired, and allows the caller to await for them to all be resolved.

**Review by derrell — APPROVED** [review]

File: source/class/qx/core/MEvent.js [hunk 1]
```diff
@@ -166,6 +166,69 @@ qx.Mixin.define("qx.core.MEvent", {
...
     },
 
+    /** @type{Object<String,qx.Promise>} list of pending events, indexed by hash code */
+    __pendingEvents: null,
+
+    /** @type{qx.Promise} promise that callers are waiting on, ready for when all events are finished */
+    __promiseWaitForPendingEvents: null,
+
+    /**
+     * Internal helper method to track promises returned from evcent handlers
```
Comment (cboulanger) [inline source/class/qx/core/MEvent.js:176]: s/evcent/event/

**Review by cboulanger — APPROVED** [review]

**Review by hkollmann — APPROVED** [review]

**Review by oetiker — APPROVED** [review]

**Review by level420 — APPROVED** [review]

**Comment by hkollmann** [comment]
Thanks @johnspackman !

**Pull request merged**